### PR TITLE
task18

### PR DESCRIPTION
**Pull request by nastyamyalik**
task18

### CODE REVIEW

File: tasks/18.txt [hunk 1]
```diff
@@ -1,3 +1,10 @@
...
-| Order Date | Total Number of Orders |
\ No newline at end of file
+| Order Date | Total Number of Orders |
+
+//task18
+db.orders.aggregate([
+     { $match: {OrderDate: { $regex: /^1998/ }}},
+     { $sort: {OrderDate:-1}},
+     { $group: { _id: "$OrderDate" , TotalNumberOfOrders : { $sum: 1 } } }
+ ])
```
Comment (ilya4321) [inline tasks/18.txt]: Please run the query before
db.orders.find().forEach(order=>{
    order.OrderDate = new Date(order.OrderDate)
    order.RequiredDate = new Date(order.RequiredDate)
    order.ShippedDate = new Date(order.ShippedDate)
    db.orders.save(order)
    })

File: tasks/18.txt [hunk 2]
```diff
@@ -4,7 +4,12 @@ Create a SQL query to calcualte total orders count by each day in 1998:
 
 //task18
 db.orders.aggregate([
+{
+    $project: {
+    OrderDate: { $dateToString: { format: "%Y-%m-%d", date: "$OrderDate" } },
+ }
+ },
```
Comment (ilya4321) [inline tasks/18.txt]: still can be faster without $project

**Comment by nastyamyalik** [comment]
I dont have the idea for now, how to fix it without $ project)
But maybe it will be faster than the previous one, like one more solution)

**Comment by ilya4321** [comment]
@nastyamyalik you can just use dates interval

**Comment by ilya4321** [comment]
@nastyamyalik the problem is mongodb dosn't use any indexes after project and can be really slow.
We have to filter as more as we can before first $project

**Pull request closed without merging**